spec/services/rate_limit_checker_spec.rb

```diff
@@ -5,11 +5,34 @@
   let(:article) { create(:article, user: user) }
   let(:rate_limit_checker) { described_class.new(user) }
 
+  def cache_key(action)
+    rate_limit_checker.send("limit_cache_key", action)
+  end
+
   describe "#limit_by_action" do
```

Comment:
we should probably add `:aggregate_failures` here as some of the expectations are created at runtime and it'd be complicated to find them otherwise, as the same line corresponds to different test cases

Reply:
wait, can we aggregate failures in a describe? RSpec is magical :D